src/invision/components/movie-popup/movie.popup.js

```diff
@@ -1,87 +1,88 @@
-import React from "react";
+import React, { useState } from "react";
 import Modal from 'react-awesome-modal';
 
-export default class MoviePopup extends React.Component {
-  constructor(props) {
-    super(props)
-    this.state = {
-      title: this.props.movie?.title || '',
-      releaseDate: new Date(this.props.movie?.releaseDate),
-      url: this.props.movie?.url || '',
-      genre: this.props.movie?.genre || '',
-      overview: this.props.movie?.overview || '',
-      runtime: this.props.movie?.runtime || ''
-    }
-    this.handleTitleChange = this.handleTitleChange.bind(this);
-    this.handleReleaseDateChange = this.handleReleaseDateChange.bind(this);
-    this.handleUrlChange = this.handleUrlChange.bind(this);
-    this.handleGenreChange = this.handleGenreChange.bind(this);
-    this.handleOverviewChange = this.handleOverviewChange.bind(this);
-    this.handleRuntimeChange = this.handleRuntimeChange.bind(this);
+export default function MoviePopup(props) {
+
+
+  const isAdd = props.title === 'Add Movie';
+
+  const [movie, setState] = useState({ ...props.movie });
+
+  const reset = () => {
+    setState({
```

Comment:
You use this MoviePopup for edit and create functionality. In this case you cannot use the same reset functionality because initial values of editing movie is not undefined. When you open edit movie, click reset and save, you get movie with enpty data.